cores/esp8266/WString.cpp

```diff
@@ -617,6 +617,19 @@ int String::indexOf(char ch, unsigned int fromIndex) const {
     return temp - buffer();
 }
 
+int String::indexOf(const char *s2) const {
+    return indexOf(s2, 0);
+}
+
+int String::indexOf(const char *s2, unsigned int fromIndex) const {
+    if (fromIndex >= len())
+        return -1;
+    const char *found = strstr(buffer() + fromIndex, s2);
```

Comment:
If you change this to `strstr_P` it will work with `PROGMEM` strings natively.

Reply:
Perfect, thanks!

Reply:
I think that @earlephilhower was suggesting that only one full body of `::indexof()` was needed if `strstr_P()` is used instead of `strstr()`.

Reply:
I see, I'm keeping both to prevent `F(...)` to be converted to String implicitly, but I will make the char* version use `strstr_P` too then. Is that cool?

Reply:
I think so !

Reply:
Done!